src/util/SpawnObject.cpp

```diff
@@ -18,12 +47,31 @@ namespace YimMenu
 				ScriptMgr::Yield();
 
 				ENTITY::SET_ENTITY_VISIBLE(obj, true);
+				ENTITY::SET_ENTITY_ROTATION(obj, pitch, roll, yaw, 2, true); // Set rotation
 
 				NETWORK::NETWORK_REGISTER_ENTITY_AS_NETWORKED(obj);
 				int id = NETWORK::OBJ_TO_NET(obj);
 				if (NETWORK::NETWORK_DOES_NETWORK_ID_EXIST(id))
 				{
-					OBJECT::PLACE_OBJECT_ON_GROUND_PROPERLY(obj, true);
+					if (onGround)
```

Comment:
For the several if statements here, just pass the bool directly to the native.
Like this: `OBJECT::PLACE_OBJECT_ON_GROUND_PROPERLY(obj,onGround);`

Reply:
Only for the On_ground native and Freeze_position native, the isBurning and isPickup wont apply since there are no actual true/false values